src/main/java/org/infernalstudios/miningmaster/recipes/GemSmithingRecipe.java

```diff
@@ -61,14 +62,20 @@ public GemSmithingRecipe(ResourceLocation recipeId, Ingredient blacklist, ItemSt
 
     @Override
     public boolean matches(Container inv, Level worldIn) {
-        return !this.blacklist.test(inv.getItem(0)) && this.gem.sameItem(inv.getItem(1)) && assemble(inv, worldIn.registryAccess()) != null;
+        return !this.blacklist.test(inv.getItem(1)) && ItemStack.isSameItem(this.gem, inv.getItem(2)) && assemble(inv, worldIn.registryAccess()) != null;
     }
 
     @Override
     public ItemStack assemble(Container inv, RegistryAccess registryAccess) {
-        ItemStack itemstack = inv.getItem(0).copy();
-        CompoundTag compoundnbt = inv.getItem(0).getTag();
+        MiningMaster.LOGGER.debug("Slot 0: " + inv.getItem(0).copy());
+        MiningMaster.LOGGER.debug("Slot 1: " + inv.getItem(0).copy());
+        MiningMaster.LOGGER.debug("Slot 2: " + inv.getItem(0).copy());
```

Comment:
these log statements should probably be `trace()` not `debug()`, since they aren't written to any logs iirc, and they should stay as developer-only logs

Reply:
removed the log statements since there wasn't any need to trace them either